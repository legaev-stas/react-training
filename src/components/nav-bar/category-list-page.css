.search-bar .icon-holder{
    transition: max-width 1.2s;
    max-width: 100px;
}
.search-bar .icon-holder.hidden{
    max-width: 0;
    transition: max-width .3s;
}