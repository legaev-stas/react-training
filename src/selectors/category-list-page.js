import {createSimpleSelector, createSelector} from '../helpers/selector';
import {getState} from '../helpers/store';


const categoryStoreSlice = () => getState().category;
const categoryCollection = createSimpleSelector(categoryStoreSlice, 'collection');
const taskStoreSlice = () => getState().task;
const taskCollection = createSimpleSelector(taskStoreSlice, 'collection');


export const categoryList = createSelector([categoryCollection, taskCollection], (categoryCollection, taskCollection) => {
    let collection = categoryCollection.map(category => {
        let tasks = taskCollection.filter(task => task.get('category') === category.get('id'));
        let badge = tasks.filter(task => !task.get('completed')).size;

        return category.merge({
            badge,
            tasks
        });
    });

    return {collection};
});