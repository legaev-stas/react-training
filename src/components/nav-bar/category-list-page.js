import React from 'react';
import {SearchBar, Flex, WingBlank, Icon} from 'antd-mobile';
import './category-list-page.css';


export class AppNavBar extends React.Component {
    // bug fixing when component doesn't show cancel button after focus
    onSetSearchModeOn = () => {
        setTimeout(this.props.onSetSearchModeOn, 0)
    }

    render() {
        return (
            <div className="search-bar">
                <Flex alignContent="center">
                    <Flex.Item>
                        <SearchBar
                            cancelText="Cancel"
                            value={this.props.searchQuery}
                            showCancelButton={this.props.searchMode}
                            onFocus={this.onSetSearchModeOn}
                            onCancel={this.props.onSetSearchModeOff}
                            onChange={this.props.onChangeSearchQuery}
                        />
                    </Flex.Item>
                    <div className={this.props.searchMode ? 'hidden icon-holder' : 'icon-holder'}>
                        <WingBlank>
                            <Icon type="plus" size="sm" onClick={this.props.addItem} color="#108ee9"/>
                        </WingBlank>
                    </div>
                </Flex>
            </div>
        );
    }
}
