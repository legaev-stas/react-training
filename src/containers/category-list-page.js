import React from 'react';
import {List} from '../components/list';
import {AppNavBar} from '../components/nav-bar/category-list-page';
import {Modal} from 'antd-mobile';


export class CategoryListPage extends React.Component {
    constructor(props) {
        super(props);

        this.editCategory = this.editCategory.bind(this);
        this.deleteCategory = this.deleteCategory.bind(this);
        this.openCategory = this.openCategory.bind(this);
    }

    deleteCategory({id, badge}) {
        const heading = 'Delete Category';
        const message = `Are you sure? The category contains ${badge} uncompleted tasks`;

        if (badge) {
            Modal.alert(heading, message, [
                {text: 'Cancel', style: 'default'},
                {text: 'OK', onPress: () => this.props.deleteCategory(id)},
            ]);
        } else {
            this.props.deleteCategory(id);
        }
    }

    openCategory({id}) {
        this.props.openCategory(id);
    }

    editCategory({id, title:originalTitle}) {
        Modal.prompt('Update category name', '',
            [
                {text: 'Cancel'},
                {
                    text: 'Update',
                    onPress: title => new Promise((resolve) => {
                        this.props.editCategory({
                            id,
                            title
                        });
                        resolve();
                    }),
                },
            ], 'default', originalTitle);
    }

    render() {
        return (
            <div>
                <AppNavBar addCategory={this.props.addCategory}/>
                <List
                    collection={this.props.collection}
                    onEdit={this.editCategory}
                    onDelete={this.deleteCategory}
                    onClick={this.openCategory}
                />
            </div>
        );
    }
}

