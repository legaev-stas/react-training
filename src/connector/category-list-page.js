import {connect} from 'react-redux';
import {categoryList} from '../selectors/category-list-page';
import {
    createCategory,
    editCategory,
    deleteCategory,
    openCategory
} from '../actions/category/index';

export const categoryListPageConnector = connect(categoryList, {
    editCategory,
    deleteCategory,
    openCategory,
    createCategory
});
