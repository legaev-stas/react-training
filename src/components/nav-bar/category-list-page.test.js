import React from 'react';

import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()});

import {AppNavBar} from './category-list-page';


it('Component should hide add category icon when it is searchMode', () => {
    const tree = shallow(<AppNavBar searchMode/>);
    expect(toJson(tree)).toMatchSnapshot();
});
it('Component should show add category icon when it is not searchMode', () => {
    const tree = shallow(<AppNavBar/>);
    expect(toJson(tree)).toMatchSnapshot();
});
