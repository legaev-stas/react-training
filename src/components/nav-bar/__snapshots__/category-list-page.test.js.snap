// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should hide add category icon when it is searchMode 1`] = `
<div
  className="search-bar"
>
  <Flex
    align="center"
    alignContent="center"
    prefixCls="am-flexbox"
  >
    <FlexItem
      prefixCls="am-flexbox"
    >
      <SearchBar
        cancelText="Cancel"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onFocus={[Function]}
        onSubmit={[Function]}
        placeholder=""
        prefixCls="am-search"
        showCancelButton={true}
      />
    </FlexItem>
    <div
      className="hidden icon-holder"
    >
      <WingBlank
        prefixCls="am-wingblank"
        size="lg"
      >
        <Icon
          color="#108ee9"
          size="sm"
          type="plus"
        />
      </WingBlank>
    </div>
  </Flex>
</div>
`;

exports[`Component should show add category icon when it is not searchMode 1`] = `
<div
  className="search-bar"
>
  <Flex
    align="center"
    alignContent="center"
    prefixCls="am-flexbox"
  >
    <FlexItem
      prefixCls="am-flexbox"
    >
      <SearchBar
        cancelText="Cancel"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClear={[Function]}
        onFocus={[Function]}
        onSubmit={[Function]}
        placeholder=""
        prefixCls="am-search"
        showCancelButton={false}
      />
    </FlexItem>
    <div
      className="icon-holder"
    >
      <WingBlank
        prefixCls="am-wingblank"
        size="lg"
      >
        <Icon
          color="#108ee9"
          size="sm"
          type="plus"
        />
      </WingBlank>
    </div>
  </Flex>
</div>
`;
